src/Parser/Template.cs

```diff
@@ -54,6 +57,95 @@ public Template(string template, bool isPositional, List<Literal> literals, List
       Holes = holes.ToArray();
     }
 
+    public string Render(object[] parameters)
```

Comment:
I don't know.

To know what is most performant would require benchmarking and it probably doesn't matter much. With a renderer class, we could store all the parameters in members, but I am unsure if that's better or worse.

I am a bit unsettled by the fact that _all the members_ of `Template` would need to be `internal` for a renderer class to work. That's kind of a smell to me that `Render` belongs to `Template`. And to be honest, what's the only goal of a template if it's not being rendered?

The one thing I would drop when we're done is the `Rebuild` method, which in my opinion is not required anymore. We will have other means to test the template.